### ISSUE TITLE
1.11.4 manual is not correct

### ISSUE THREAD

**Issue by fubarbaz** [NixOS/nix#1170]
The problem is in `14.4. Building and Testing` of the Nix manual

Expected result:

![image](https://cloud.githubusercontent.com/assets/23479333/21603997/1923d5ce-d1a1-11e6-9308-efb97df9fb7d.png)

Obtained result
```
$ file pkgs/top-level/all-packages.nix 
pkgs/top-level/all-packages.nix: UTF-8 Unicode text
$ /run/current-system/sw/bin/nix-build pkgs/top-level/all-packages.nix -A hello
error: cannot auto-call a function that has an argument without a default value (‘system’)
$ nix-build --version
nix-build (Nix) 1.11.4
```
![image](https://cloud.githubusercontent.com/assets/23479333/21603953/df5aaf98-d1a0-11e6-9a67-d3d922b16149.png)

**Comment by edolstra** [comment]
Fixed a while ago in 96d3534a9ed0d93c5e819543692933e50f246982.

**Comment by edolstra** [comment]
Cherry-picked to 1.11-maintenace.
